File: 119.Pascal-sTriangle-II/clywin123/getRow.py
class Solution:
    # @param {integer} rowIndex
    # @return {integer[]}
    def getRow(self, rowIndex):
        if(rowIndex==0):
            return [1]
        elif(rowIndex==1):
            return [1,1]
        else:
            answer = [1,1]
            for i in range(1,rowIndex):
                for j in range(len(answer)-1):
                    answer.append(answer[j]+answer[j+1])
                answer =  answer[j+2:]
                answer = [1]+answer+[1]
            return answer
        

File: 119.Pascal-sTriangle-II/wyrj/getRow.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    const result = new Array(rowIndex + 1);
    for (let i = 0; i <= rowIndex; i++) {
        result[i] = 1;
        for (let j = i - 1; j > 0; j--) {
            result[j] += result[j - 1];
        }
    }
    return result;
};
